static/app/views/settings/organizationGeneralSettings/organizationSettingsForm.tsx

```diff
@@ -20,47 +16,40 @@ import organizationSettingsFields from 'sentry/data/forms/organizationGeneralSet
 import {IconCodecov, IconLock} from 'sentry/icons';
 import {t} from 'sentry/locale';
 import {space} from 'sentry/styles/space';
-import type {Organization, OrganizationAuthProvider, Scope} from 'sentry/types';
-import withOrganization from 'sentry/utils/withOrganization';
+import type {Organization} from 'sentry/types';
+import {useLocation} from 'sentry/utils/useLocation';
+import useOrganization from 'sentry/utils/useOrganization';
 
 const HookCodecovSettingsLink = HookOrDefault({
   hookName: 'component:codecov-integration-settings-link',
 });
 
-interface Props extends RouteComponentProps<{}, {}> {
-  access: Set<Scope>;
+interface Props {
   initialData: Organization;
-  location: Location;
   onSave: (previous: Organization, updated: Organization) => void;
-  organization: Organization;
 }
 
-interface State extends AsyncComponentState {
-  authProvider: OrganizationAuthProvider;
-}
-
-class OrganizationSettingsForm extends DeprecatedAsyncComponent<Props, State> {
-  getEndpoints(): ReturnType<DeprecatedAsyncComponent['getEndpoints']> {
-    const {organization} = this.props;
-    return [['authProvider', `/organizations/${organization.slug}/auth-provider/`]];
-  }
+function OrganizationSettingsForm({initialData, onSave}: Props) {
+  const location = useLocation();
+  const organization = useOrganization();
+  const endpoint = `/organizations/${organization.slug}/`;
 
-  render() {
-    const {initialData, organization, onSave, access} = this.props;
-    const {authProvider} = this.state;
-    const endpoint = `/organizations/${organization.slug}/`;
+  const access = useMemo(() => new Set(organization.access), [organization]);
 
-    const jsonFormSettings = {
-      additionalFieldProps: {hasSsoEnabled: !!authProvider},
```

Comment:
`hasSsoEnabled` is not used anymore. It seems this a leftover from when security settings were included in this form.